python/tests/test_types.py

```diff
@@ -15,149 +15,541 @@
+        (List[Map[String, Integer]], List[Map[String, Number]], True),
+        (List[Map[String, Integer]], List[Map[Float, Number]], False),
+        (List[Map[String, Decimal[32, 3]]], List[Map[String, Number]], True),
+        (List, Map, False),
+        (Double, IcebergType, True),
+        (Float, Number, True),
+        (Long, Fixed, False),
+    ],
+)
+def test_issubclass(type1, type2, result):
+    assert issubclass(type1, type2) == result
+
+
+@pytest.mark.parametrize(
+    "type, result",
+    [
+        (List[Integer], List),
+        (Map[String, Integer], Map),
+        (List[Integer], List),
+        (List[Map[String, Integer]], List),
+        (List[Map[String, Decimal[32, 3]]], List),
+        (NestedField[Double, False, 2, "required_field"], NestedField),
+        (List, List),
+        (Decimal, Decimal),
+        (Fixed[8], Fixed),
+        (Decimal[16, 3], Decimal),
+        (Double, None),
+        (Float, None),
+        (Long, None),
+    ],
+)
+def test_get_unspecified_generic_type(type, result):
+    if result is None:
+        with pytest.raises(TypeError):
+            generic_class.get_unspecified_generic_type(type)
+    else:
+        assert generic_class.get_unspecified_generic_type(type) == result
+
+
+@pytest.mark.parametrize(
+    "type_, r1, r2, r3",
+    [
+        (List[Integer], False, True, True),
+        (Map[String, Integer], False, True, True),
+        (List[Integer], False, True, True),
+        (List[Map[String, Integer]], False, True, True),
+        (List[Map[String, Decimal[32, 3]]], False, True, True),
+        (NestedField[String, False, 2, "required_field"], False, True, True),
+        (List, True, True, False),
+        (Decimal, True, True, False),
+        (Fixed[8], False, True, True),
+        (Decimal[16, 3], False, True, True),
+        (Double, False, False, False),
+        (Float, False, False, False),
+        (Long, False, False, False),
+    ],
+)
+def test_is_generic_type(type_, r1, r2, r3):
+    assert generic_class.is_generic_type(type_) == r1
+    assert generic_class.is_generic_type(type_, True) == r2
+    assert generic_class.is_generic_type(type_, True, True) == r3
+
+
+@pytest.mark.parametrize(
+    "order",
+    [
+        Float("-inf")
+        < Float("-nan")
+        < Float(-4e10)
+        < Float(-24)
+        < Float("-0")
+        < Float(0)
+        < Float(24)
+        < Float(4e10)
+        < Float("nan")
+        < Float("inf"),
+        Float("-inf")
+        <= Float("-nan")
+        <= Float(-4e10)
+        <= Float(-24)
+        <= Float("-0")
+        <= Float(0)
+        <= Float(24)
+        <= Float(4e10)
+        <= Float("nan")
+        <= Float("inf"),
+        Double("-inf")
+        < Double("-nan")
+        < Double(-4e10)
+        < Double(-24)
+        < Double("-0")
+        < Double(0)
+        < Double(24)
+        < Double(4e10)
+        < Double("nan")
+        < Double("inf"),
+        Double("-inf")
+        <= Double("-nan")
```

Comment:
I thin this violates the order required by the spec: -NaN < -Infinity < ...

See https://iceberg.apache.org/#spec/#sorting

Reply:
Thanks a lot. I think it's fixed I just misread the spec